types/aws-lambda/index.d.ts

```diff
@@ -190,6 +197,7 @@ interface ProxyResult {
  */
 export type Handler = (event: any, context: Context, callback?: Callback) => void;
 export type ProxyHandler = (event: APIGatewayEvent, context: Context, callback?: ProxyCallback) => void;
+export type CustomAuthorizerHandler = (event: CustomAuthorizerEvent, context: Context, callback?: Callback) => void;
```

Comment:
I think you should define a custom type for the callback parameter. That way you can check the structure passed to the callback is valid for a custom authorizer. See [Output from an Amazon API Gateway Custom Authorizer](http://docs.aws.amazon.com/apigateway/latest/developerguide/use-custom-authorizer.html) and how it's done for the ProxyHandler